editoast/src/views/timetable.rs

```diff
@@ -396,27 +398,30 @@ pub struct Conflict {
 impl Conflict {
     /// This function processes train ids from Core Response
     ///  and maps them to either a `train_schedule_id` or a `paced_train_occurrence_id` based on the provided key mapping.
-    fn from_core_response(conflict: CoreConflict) -> Result<Self> {
+    fn from_core_response(
+        conflict: CoreConflict,
+        trains_map: &HashMap<String, TrainId>,
+    ) -> Result<Self> {
         let (train_schedule_ids, paced_train_occurrence_ids): (Vec<_>, Vec<_>) = conflict
             .train_ids
             .iter()
-            .partition_map(|train_id| match train_id.parse() {
-                Ok(TrainId::TrainSchedule(id)) => Either::Left(id),
-                Ok(TrainId::PacedTrain {
+            .partition_map(|train_id| match trains_map.get(train_id).cloned() {
```

Comment:
Since we’re both dealing with UUID coming back from `core` and `TrainId` from the `HashMap`, I suggest to be more explicit here in the name of the variables.

```suggestion
            .partition_map(|train_uuid| match trains_map.get(train_uuid).cloned() {
```